lib/auth/webauthnwin/api.go

```diff
@@ -33,10 +44,207 @@ const (
+	if origin == "" {
+		return nil, trace.BadParameter("origin required")
+	}
+	if err := cc.Validate(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	rp, err := rpToCType(cc.Response.RelyingParty)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	u, err := userToCType(cc.Response.User)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	credParam, err := credParamToCType(cc.Response.Parameters)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	cd, jsonEncodedCD, err := clientDataToCType(cc.Response.Challenge.String(), origin, string(protocol.CreateCeremony))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	opts, err := makeCredOptionsToCType(cc.Response)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	resp, err := native.MakeCredential(origin, &makeCredentialRequest{
+		rp:                    rp,
+		user:                  u,
+		credParameters:        credParam,
+		clientData:            cd,
+		jsonEncodedClientData: jsonEncodedCD,
+		opts:                  opts,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &proto.MFARegisterResponse{
+		Response: &proto.MFARegisterResponse_Webauthn{
+			Webauthn: wanlib.CredentialCreationResponseToProto(resp),
+		},
+	}, nil
+}
+
 // CheckSupport is the result from a Windows webauthn support check.
 type CheckSupportResult struct {
 	HasCompileSupport  bool
 	IsAvailable        bool
 	HasPlatformUV      bool
 	WebAuthnAPIVersion int
 }
+
+// IsAvailable returns true if Windows webauthn library is available in the
+// system. Typically, a series of checks is performed in an attempt to avoid
+// false positives.
+// See CheckSupport.
+func IsAvailable() bool {
+	return CheckSupport().IsAvailable
+}
+
+// CheckSupport return information whether Windows Webauthn is supported and
+// information about API version.
+func CheckSupport() CheckSupportResult {
+	return native.CheckSupport()
+}
+
+type DiagResult struct {
+	Available                           bool
+	RegisterSuccessful, LoginSuccessful bool
+}
+
+// Diag runs a few diagnostic commands and returns the result.
+// User interaction is required.
+func Diag(ctx context.Context, promptOut io.Writer) (*DiagResult, error) {
+	res := &DiagResult{}
+	if !IsAvailable() {
+		return res, nil
+	}
+	res.Available = true
+
+	// Attempt registration.
+	const origin = "localhost"
+	cc := &wanlib.CredentialCreation{
+		Response: protocol.PublicKeyCredentialCreationOptions{
+			Challenge: make([]byte, 32),
+			RelyingParty: protocol.RelyingPartyEntity{
+				ID: "localhost",
+				CredentialEntity: protocol.CredentialEntity{
+					Name: "test RP",
+				},
+			},
+			User: protocol.UserEntity{
+				CredentialEntity: protocol.CredentialEntity{
+					Name: "test",
+				},
+				ID:          []byte("test"),
+				DisplayName: "test",
+			},
+			Parameters: []protocol.CredentialParameter{
+				{
+					Type:      protocol.PublicKeyCredentialType,
+					Algorithm: webauthncose.AlgRS256,
+				},
```

Comment:
```suggestion
				{
					Type:      protocol.PublicKeyCredentialType,
					Algorithm: webauthncose.AlgES256,
				},
				{
					Type:      protocol.PublicKeyCredentialType,
					Algorithm: webauthncose.AlgRS256,
				},
```

In case someone wants to use a FIDO2/U2F device.